hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCreateRequest.java

```diff
@@ -213,10 +213,9 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
 
       OmBucketInfo bucketInfo = omMetadataManager.getBucketTable().get(
           omMetadataManager.getBucketKey(volumeName, bucketName));
-      encryptionInfo = getFileEncryptionInfo(ozoneManager, bucketInfo);
 
       omKeyInfo = prepareKeyInfo(omMetadataManager, keyArgs, dbKeyInfo,
-          keyArgs.getDataSize(), locations, encryptionInfo.orNull(),
+          keyArgs.getDataSize(), locations,  getFileEncryptionInfo(keyArgs),
```

Comment:
NIT: Given we have pass keyArgs, we can reduce the number of arguments here to  avoid @SuppressWarnings("parameternumber") for prepareKeyInfo().

Reply:
By removing the encryptionInfo and size, we cannot remove @SuppressWarnings("parameternumber"). I will leave for now. I will open a new jira to refactor this.

Reply:
Jira to track this
https://issues.apache.org/jira/browse/HDDS-3407

Reply:
Ok to me.  Let's fix that later. +1.